examples/vision/edsr.py

```diff
@@ -0,0 +1,276 @@
+    )
+
+    hr_width = lr_width * scale
+    hr_height = lr_height * scale
+
+    lr_img_cropped = lr_img[
+        lr_height : lr_height + lr_crop_size, lr_width : lr_width + lr_crop_size
+    ]
+    hr_img_cropped = hr_img[
+        hr_height : hr_height + hr_crop_size, hr_width : hr_width + hr_crop_size
+    ]
+
+    return lr_img_cropped, hr_img_cropped
+
+
+"""
+# Loading Data using tensorflow datasets
+"""
+
+div2k_data = tfds.image.Div2k(config="bicubic_x4")
+div2k_data.download_and_prepare()
+
+train = div2k_data.as_dataset(split="train", as_supervised=True)
+train_cache = train.cache()
+
+ds = train_cache
+
+ds = ds.map(lambda lr, hr: random_crop(lr, hr, scale=4), num_parallel_calls=AUTOTUNE)
+ds = ds.map(random_rotate, num_parallel_calls=AUTOTUNE)
+ds = ds.map(flip_left_right, num_parallel_calls=AUTOTUNE)
+
+ds = ds.batch(16)
+ds = ds.repeat(None)
+ds = ds.prefetch(buffer_size=AUTOTUNE)
+
+lr, hr = next(iter(ds))
+
+"""
+# High Resolution Images
+"""
+
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(hr[i].numpy().astype("uint8"))
+    plt.title(hr[i].shape)
+    plt.axis("off")
+
+"""
+# Low Resolution Images
+"""
+
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(lr[i].numpy().astype("uint8"))
+    plt.title(lr[i].shape)
+    plt.axis("off")
+
+"""
+# Normalization and usefull functions
+As, the paper mentioned we will subtract image by mean rgb of div2k dataset
+"""
+
+DIV2K_RGB_MEAN = np.array([0.4488, 0.4371, 0.4040]) * 255
+
+
+def normalize(x, rgb_mean=DIV2K_RGB_MEAN):
+    return (x - rgb_mean) / 127.5
+
+
+def denormalize(x, rgb_mean=DIV2K_RGB_MEAN):
+    return (x * 127.5) + rgb_mean
+
+
+def shuffle_pixels(scale):
+    return lambda x: tf.nn.depth_to_space(x, scale)
+
+
+# computes the peak signal-to-noise ratio, measures quality of image
+def PSNR(sr, hr):
+    psnr_value = tf.image.psnr(hr, sr, max_val=255)[0]
+    return psnr_value
+
+
+"""
+# Model Building(Baseline Model)
+1. 16 residual blocks
+2. filters=64
+3. No batchNormalization
+"""
+
+"""
+<img src = "https://miro.medium.com/max/1400/1*a1tQ4grku_ugTTe39IJzjA.png" width="500">
+"""
+
+
+def ResBlock(inputs):
+    x = tf.keras.layers.Conv2D(64, 3, padding="same", activation="relu")(inputs)
+    x = tf.keras.layers.Conv2D(64, 3, padding="same")(x)
+    x = tf.keras.layers.Add()([inputs, x])
+    return x
+
+
+def Upsampling(inputs, factor=2, **kwargs):
+    x = tf.keras.layers.Conv2D(64 * (factor**2), 3, padding="same", **kwargs)(inputs)
+    x = tf.keras.layers.Lambda(shuffle_pixels(scale=factor))(x)
+    x = tf.keras.layers.Conv2D(64 * (factor**2), 3, padding="same", **kwargs)(x)
+    x = tf.keras.layers.Lambda(shuffle_pixels(scale=factor))(x)
+    return x
+
+
+def EDSR():
+    input_layer = tf.keras.layers.Input(shape=(None, None, 3))
+    x = tf.keras.layers.Lambda(normalize)(input_layer)
```

Comment:
Don't use a `Lambda` layer. Instead, use a `Rescaling` layer to scale the input values.